lib/pavilion/builder.py

```diff
@@ -608,6 +612,16 @@ def _setup_build_dir(self, dest):
                     "Could not copy extra file '{}' to dest '{}': {}"
                     .format(path, dest, err))
 
+    def make_files(self, files, dst):
+        for file in files:
+            # Assumes that each file is a path, e.g., 'somedir/file', or './file'
+            dirname = os.path.dirname(file)
+            Path(dst / dirname).mkdir(parents=True, exist_ok=True)
+            path = Path(file)
+            with open(dst / path, 'w') as f:
+                for line in file:
+                    f.write(line)
```

Comment:
I'm pretty sure you're writing the filename to file here, not the lines in the file.